File: app/src/main/java/mariavv/fitnesspal/presentation/journal/daycard/DayCardView.java
package mariavv.fitnesspal.presentation.journal.daycard;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.SingleStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

import java.util.List;

@StateStrategyType(SingleStateStrategy.class)
interface DayCardView extends MvpView {
    void updateCard(List<ItemType> dataSet, int dayProtein, int dayFat, int dayCarb, int dayEnergy);
}

File: app/src/main/java/mariavv/fitnesspal/presentation/journal/daycard/DayCardView.kt
package mariavv.fitnesspal.presentation.journal.daycard

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.SingleStateStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

@StateStrategyType(SingleStateStrategy::class)
interface DayCardView : MvpView {
    fun updateCard(dataSet: List<ItemType>, dayProtein: Int, dayFat: Int, dayCarb: Int, dayEnergy: Int)
}
